pandas/core/generic.py

```diff
@@ -8970,17 +8970,17 @@ def _find_valid_index(self, how):
 
         if how == 'first':
             # First valid value case
-            i = is_valid.idxmax()
-            if not is_valid[i]:
-                return None
-            return i
+            i = is_valid.values[::].argmin()
```

Comment:
just call this idxpos, no need for i any longer

Reply:
Thanks - done